### ISSUE TITLE
The response is incomplete when throw exception in resolver.

### ISSUE THREAD

**Issue by jaggerwang** [graphql-java-kickstart/graphql-spring-boot#304]
## Description

When I throw `UnauthenticatedException` in resolver, the exception stack is printed on the console like not handled(already enabled `exception-handlers-enabled`). And I got the following incomplete response.
![image](https://user-images.githubusercontent.com/1255011/70112300-d6512d00-168f-11ea-8afb-5d59afcacdf9.png)

```java
@Component
public class QueryResolver implements GraphQLQueryResolver {
    ...

    public UserEntity userInfo(Long id) {
        throw new UnauthenticatedException("未认证");
        // return userUsecases.info(id);
    }

    ...
}
```

The same problem shows up when I using `ExceptionHandler` to handle exception. I want return my custom type code according to the exception type, but there only have message filed in errors, no type field. Also I want to known is it possible to set the response http status code?

```java
public class CustomGraphQLError extends ThrowableGraphQLError {
    private static final long serialVersionUID = 1L;

    private String type;

    public CustomGraphQLError(Throwable throwable, String message, String type) {
        super(throwable, message);

        this.type = type;
    }

    @Override
    public String getType() {
        return type == null ? super.getType() : type;
    }
}
```

```java
@Component
public class GraphQLExceptionHandlers {
    @ExceptionHandler(Exception.class)
    public CustomGraphQLError handle(Exception exception) {
        exception.printStackTrace();

        return new CustomGraphQLError(exception, exception.toString(), "fail");
    }

    @ExceptionHandler(UnauthenticatedException.class)
    public CustomGraphQLError handle(UnauthenticatedException exception) {
        return new CustomGraphQLError(exception, exception.getMessage(), "unauthenticated");
    }
}
```

## Env

**application.yml**
```yaml
spring:
    application:
        name: spring-boot-in-practice
    servlet:
        multipart:
            file-size-threshold: 10MB
            location: /tmp
            max-file-size: 100MB
            max-request-size: 100MB
    datasource:
        url: ${SBIP_SPRING_DATASOURCE_URL:jdbc:mysql://localhost/sbip}
        username: ${SBIP_SPRING_DATASOURCE_USERNAME:root}
        password: ${SBIP_SPRING_DATASOURCE_PASSWORD:}
    flyway:
        enabled: true
        clean-disabled: true
        baseline-on-migrate: true
    redis:
        host: ${SBIP_SPRING_REDIS_HOST:localhost}
        port: ${SBIP_SPRING_REDIS_PORT:6379}
        password: ${SBIP_SPRING_REDIS_PASSWORD:}
        timeout: 1s
        lettuce:
            pool:
                max-wait: 1s
    session:
        timeout: 7d
        store-type: redis
        redis:
            namespace: sbip:session
    jpa:
        open-in-view: false

graphql:
    servlet:
        enabled: true
        corsEnabled: true
        mapping: /graphql
        exception-handlers-enabled: true
        tracingEnabled: false
        actuator-metrics: true

graphiql:
    enabled: true
    mapping: /graphiql
    cdn:
        enabled: false

voyager:
    enabled: true
    cdn:
        enabled: false
```

**pom.xml**
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.2.1.RELEASE</version>
		<relativePath /> <!-- lookup parent from repository -->
	</parent>
	<groupId>net.jaggerwang</groupId>
	<artifactId>spring-boot-in-practice</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<name>Sbip</name>
	<description>Spring boot in practice</description>

	<properties>
		<java.version>11</java.version>
		<start-class>net.jaggerwang.sbip.SbipApplication</start-class>
	</properties>

	<dependencies>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-data-jpa</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-data-redis</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-security</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.session</groupId>
			<artifactId>spring-session-data-redis</artifactId>
		</dependency>

		<dependency>
			<groupId>com.graphql-java-kickstart</groupId>
			<artifactId>graphql-spring-boot-starter</artifactId>
			<version>6.0.0</version>
		</dependency>
		<dependency>
			<groupId>com.graphql-java-kickstart</groupId>
			<artifactId>graphiql-spring-boot-starter</artifactId>
			<version>6.0.0</version>
		</dependency>
		<dependency>
			<groupId>com.graphql-java-kickstart</groupId>
			<artifactId>voyager-spring-boot-starter</artifactId>
			<version>6.0.0</version>
		</dependency>

		<dependency>
			<groupId>org.projectlombok</groupId>
			<artifactId>lombok</artifactId>
		</dependency>
		<dependency>
			<groupId>mysql</groupId>
			<artifactId>mysql-connector-java</artifactId>
			<scope>runtime</scope>
		</dependency>
		<dependency>
			<groupId>org.flywaydb</groupId>
			<artifactId>flyway-core</artifactId>
		</dependency>
		<dependency>
			<groupId>org.apache.commons</groupId>
			<artifactId>commons-pool2</artifactId>
		</dependency>
		<dependency>
			<groupId>com.google.guava</groupId>
			<artifactId>guava</artifactId>
			<version>28.1-jre</version>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-actuator</artifactId>
			<optional>true</optional>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-devtools</artifactId>
			<scope>runtime</scope>
			<optional>true</optional>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-configuration-processor</artifactId>
			<optional>true</optional>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
			<exclusions>
				<exclusion>
					<groupId>org.junit.vintage</groupId>
					<artifactId>junit-vintage-engine</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
		<dependency>
			<groupId>com.graphql-java-kickstart</groupId>
			<artifactId>graphql-spring-boot-starter-test</artifactId>
			<version>6.0.0</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>com.h2database</groupId>
			<artifactId>h2</artifactId>
			<scope>test</scope>
		</dependency>
	</dependencies>

	<build>
		<plugins>
			<plugin>
				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
			</plugin>
		</plugins>
	</build>

</project>
```

**Comment by jaggerwang** [comment]
Repository: https://github.com/jaggerwang/spring-boot-in-practice

![image](https://user-images.githubusercontent.com/1255011/70382970-5df4af80-19a0-11ea-9c94-0ce9325fa9b1.png)

**Comment by elielsonms** [comment]
The message is truncated even on the RAW response? Might be an encoding breaking on Browser HTML rendering

**Comment by oliemansm** [comment]
The response being truncated was a bug that has been resolved in the latest version of this library. 

The custom error type is caused by a change that errors are serialized according to specification, which means those additional fields aren't serialized by default. If you want to create custom error that is according to spec you should override the `GraphQLError.getExtensions()` method. It's default implementation is:

```java
default Map<String, Object> getExtensions() {
  return null;
}
```

So for your example it would be something like:
```java
Map<String, Object> getExtensions() {
  Map<String, Object> extensions = new HashMap<>();
  extensions.put("type", type);
  return extensions;
}
```

We'll need to update the samples and documentation accordingly.

**Comment by jaggerwang** [comment]
Got it!

**Comment by haroldespitia** [comment]
Hi! I know this issue is now closed, but I just have a little question, I'm getting exactly the same problem described here, but after lot of experimenting using this library I realized that though the bug is correctly fixed in the current version, if you send some kind of "special characters" like question marks or accent marks you get the same error. In my particular case, I speak spanish and when I try to put some of my language characters I get this error... There's any consideration I'm mistaking?
